packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -414,9 +417,23 @@ class _PagePosition extends ScrollPositionWithSingleContext implements PageMetri
     }
     final bool result = super.applyViewportDimension(viewportDimension);
     final double oldPixels = pixels;
-    final double page = (oldPixels == null || oldViewportDimensions == 0.0) ? _pageToUseOnStartup : getPageFromPixels(oldPixels, oldViewportDimensions);
+    double page;
+    if (oldPixels == null) {
+      page = _pageToUseOnStartup;
+    } else if (oldViewportDimensions == 0.0) {
+      // If resize from zero, we should use the _cachePage to recover the state.
+      assert(_cachePage != null);
+      page = _cachePage;
+    } else {
+      page = getPageFromPixels(oldPixels, oldViewportDimensions);
+    }
+
     final double newPixels = getPixelsFromPage(page);
 
+    // Cache the page when resize the viewport to zero.
+    // For use when resize the viewport to non-zero next time.
+    _cachePage = (viewportDimension == 0.0) ? page : null;
```

Comment:
In another extreme case, assuming that the viewport is very small, only 1 pixel, I ignore the processing of this kind of scene.